src/openapi/transport/auth.js

```diff
@@ -227,77 +70,30 @@ function onTransportError(oldTokenExpiry, result) {
- * @param {string} [options.tokenRefreshCredentials="include"] - The credentials option passed to fetch.
- * @param {string} [options.tokenRefreshMethod="POST"] - The http method for refreshing the token e.g. "POST" or "GET" etc.
- * @param {number} [options.tokenRefreshMarginMs=0] - The number of ms that the refresh call should be done early.
- *          e.g. if this is 1000 it will be refreshed 1000 ms before the token expires.
- * @param {string} [options.tokenRefreshPropertyNameToken="token"] - The property name of the token after doing a refresh.
- * @param {string} [options.tokenRefreshPropertyNameExpires="expiry"] - The property name of the relative expiry after doing a refresh.
- * @param {string} [options.token] - The token to use for authentication.
- * @param {string|number} [options.expiry] - The expiry of that token, assumed to be absolute.
- * @param {number} [options.retryDelayMs] - The delay before retrying auth
- * @param {number} [options.maxRetryCount] - The maximum number of times to retry the auth url
+ * @param {number} [options.authErrorsCleanupDebounce] - The debounce timeout (in ms) used for clearing of authorization errors count.
  * @param {number} [options.maxAuthErrors] - The maximum number of authorization errors that
  *          can occur for specific endpoint within specific timeframe.
- * @param {number} [options.authErrorsCleanupDebounce] - The debounce timeout (in ms) used for clearing of authorization errors count.
  */
-function TransportAuth(baseUrl, options) { // eslint-disable-line complexity
-
-    emitter.mixinTo(this);
-
-    this.tokenRefreshUrl = options && options.tokenRefreshUrl;
-    this.tokenRefreshHeaders = options && options.tokenRefreshHeaders;
-    this.tokenRefreshCredentials = options && options.tokenRefreshCredentials || DEFAULT_TOKEN_REFRESH_CREDDENTIALS;
-    this.tokenRefreshMethod = options && options.tokenRefreshMethod || DEFAULT_TOKEN_REFRESH_METHOD;
-    this.tokenRefreshPropertyNameToken = options && options.tokenRefreshPropertyNameToken || DEFAULT_TOKEN_REFRESH_PROPERTY_NAME_TOKEN;
-    this.tokenRefreshPropertyNameExpires = options && options.tokenRefreshPropertyNameExpires || DEFAULT_TOKEN_REFRESH_PROPERTY_NAME_EXPIRES;
-    this.tokenRefreshMarginMs = options && options.tokenRefreshMarginMs || DEFAULT_TOKEN_REFRESH_MARGIN_MS;
-    this.retryDelayMs = options && options.retryDelayMs || DEFAULT_RETRY_DELAY_MS;
-    this.maxRetryCount = options && options.maxRetryCount || DEFAULT_MAX_RETRY_COUNT;
-    this.maxAuthErrors = options && options.maxAuthErrors || DEFAULT_MAX_AUTH_ERRORS;
+function TransportAuth(baseUrl, authProvider, options) {
+    if (!authProvider) {
+        throw new Error('transport auth created without a auth provider');
+    }
+
     this.authErrorsCleanupDebounce = options && options.authErrorsCleanupDebounce || DEFAULT_AUTH_ERRORS_CLEANUP_DEBOUNCE;
+    this.maxAuthErrors = options && options.maxAuthErrors || DEFAULT_MAX_AUTH_ERRORS;

```

Comment:
I wonder if we should increase default value of DEFAULT_MAX_AUTH_ERRORS. Currently it's only 3 retries which happen one after another. On the other hand, not sure if we want to ddos our servers/balancers with retries.

Reply:
We override it to Infinite and have done a long time. Not sure about the library default